src/node_embedding_api.h

```diff
@@ -0,0 +1,398 @@
+  // embedders, because it's possible for the Node.js esm loader to conflict
+  // with another one in an embedder environment, e.g. Blink's in Chromium.
+  node_embedding_runtime_no_register_esm_loader = 1 << 3,
+  // Set this flag to make Node.js track "raw" file descriptors, i.e. managed
+  // by fs.open() and fs.close(), and close them during
+  // node_embedding_delete_runtime().
+  node_embedding_runtime_track_unmanaged_fds = 1 << 4,
+  // Set this flag to force hiding console windows when spawning child
+  // processes. This is usually used when embedding Node.js in GUI programs on
+  // Windows.
+  node_embedding_runtime_hide_console_windows = 1 << 5,
+  // Set this flag to disable loading native addons via `process.dlopen`.
+  // This environment flag is especially important for worker threads
+  // so that a worker thread can't load a native addon even if `execArgv`
+  // is overwritten and `--no-addons` is not specified but was specified
+  // for this Environment instance.
+  node_embedding_runtime_no_native_addons = 1 << 6,
+  // Set this flag to disable searching modules from global paths like
+  // $HOME/.node_modules and $NODE_PATH. This is used by standalone apps that
+  // do not expect to have their behaviors changed because of globally
+  // installed modules.
+  node_embedding_runtime_no_global_search_paths = 1 << 7,
+  // Do not export browser globals like setTimeout, console, etc.
+  node_embedding_runtime_no_browser_globals = 1 << 8,
+  // Controls whether or not the Environment should call V8Inspector::create().
+  // This control is needed by embedders who may not want to initialize the V8
+  // inspector in situations where one has already been created,
+  // e.g. Blink's in Chromium.
+  node_embedding_runtime_no_create_inspector = 1 << 9,
+  // Controls whether or not the InspectorAgent for this Environment should
+  // call StartDebugSignalHandler. This control is needed by embedders who may
+  // not want to allow other processes to start the V8 inspector.
+  node_embedding_runtime_no_start_debug_signal_handler = 1 << 10,
+  // Controls whether the InspectorAgent created for this Environment waits for
+  // Inspector frontend events during the Environment creation. It's used to
+  // call node::Stop(env) on a Worker thread that is waiting for the events.
+  node_embedding_runtime_no_wait_for_inspector_frontend = 1 << 11
+} node_embedding_runtime_flags;
+
+typedef enum {
+  // Run the event loop until it is completed.
+  // It matches the UV_RUN_DEFAULT behavior.
+  node_embedding_event_loop_run_default = 0,
+  // Run the event loop once and wait if there are no items.
+  // It matches the UV_RUN_ONCE behavior.
+  node_embedding_event_loop_run_once = 1,
+  // Run the event loop once and do not wait if there are no items.
+  // It matches the UV_RUN_NOWAIT behavior.
+  node_embedding_event_loop_run_nowait = 2,
+} node_embedding_event_loop_run_mode;
```

Comment:
For embedders, I think it would be saner to just expose uv loops allowing finer grained customization.

Reply:
I am not sure about it. Please see my another comment below.